MekHQ/src/mekhq/campaign/unit/Unit.java

```diff
@@ -1138,10 +1215,158 @@ public double getInsulatedCargoCapacity() {
         }
         return capacity;
     }
+    
+    /** 
+     * Convenience method to call the right capacity getter based on unit type and weight
+     * @param unitType integer obtained from a unit's entity that denotes its type (mech, tank, etc)
+     * @param unitWeight double Weight in tons of the unit's entity. Important for tanks and infantry
+     */
+    public double getCorrectBayCapacity(int unitType, double unitWeight) {
+        final String METHOD_NAME = "getCorrectBayCapacity(unitType,unitWeight)"; //$NON-NLS-1$
+        switch (unitType) {
+            case UnitType.MEK:
+                return getCurrentMechCapacity();
+        case UnitType.AERO:
+        case UnitType.CONV_FIGHTER:
+            // Return a small craft slot if no ASF slots exist
+            if (getCurrentASFCapacity() > 0) {
```

Comment:
Do we want just add these instead?

Reply:
No I don't think we do.  I'm trying to allow for the possibility that we want to load ASFs onto a ship that has only small craft bays.  There are several of those, especially in the Pocket Warship class